site/content/languages/tinygo.md

```diff
@@ -165,19 +258,85 @@ in [runtime_wasm_wasi.go][13]. `syscall.Chdir` is implemented with the same
 [syscall_libc.go][14] used for other architectures, while `syscall.ReadDirent`
 is stubbed (returns `syscall.ENOSYS`), in [syscall_libc_wasi.go][15].
 
+## Concurrency
+
+Current versions of the WebAssembly specification do not support parallelism,
+such as threads or atomics needed to safely work in parallel.
+
+TinyGo, however, supports goroutines by default and acts like `GOMAXPROCS=1`.
+
+For example, the following code will run with the expected output, even if
+the goroutines are defined in opposite dependency order.
+```go
+package main
+
+import "fmt"
+
+func main() {
+	msg := make(chan int)
+	finished := make(chan int)
+	go func() {
+		<-msg
+		fmt.Println("consumer")
+		finished <- 1
+	}()
+	go func() {
+		fmt.Println("producer")
+		msg <- 1
+	}()
+	<-finished
+}
+```
+
+However, creating goroutines after main (`_start` in WASI) has undefined
+behavior. For example, if that same function was exported (`//export:notMain`),
+and called after main, the line that creates a goroutine panics at runtime.
+
+Given problems like this, some choose a compile-time failure instead, via the
+`-scheduler=none`. Since code often needs to be custom in order to work with
+wasm anyway, there may be limited effect to removing goroutine support.
+
+## Optimizations
+
+Below are some commonly used configurations that allow optimizing for size or
+performance vs defaults. Note that sometimes one sacrifices the other.
+
+### Binary size
+
+Those who aren't concerned about binary (`%.wasm` file) size should not
```

Comment:
```suggestion
Those who are concerned about binary (`%.wasm` file) size should not
```

Reply:
Or perhaps it was about being "extremely converned" about binary size. But this didn't read well for me

Reply:
confusing double-negative I made. I'll add fixing that to another change

Reply:
maybe

> Those unconcerned about binary (`%.wasm` file) size should use TinyGo's
defaults. Those with size constraints can reduce the wasm binary size by
changing `tinygo` flags. This can result in binaries less than 100KB.

Reply:
I think my tension is with the fact that those unconcerned about binary size may be best using `-O2`.

Maybe

> TinyGo's defaults produce fairly small binaries. Those particularly concerned with binary size can make them even smaller by...

Reply:
I guess as  defaults can approach a meg for simple code, I have some difficulty saying they are small.

I'll just ax the intro sentence and say the latter.